csharp/src/Microsoft.ML.OnnxRuntime/Training/TrainingSession.shared.cs

```diff
@@ -62,57 +71,86 @@ public class TrainingSession : IDisposable
         {
             Init(null, state, NativeOnnxValueHelper.GetPlatformSerializedString(trainModelPath), NativeOnnxValueHelper.GetPlatformSerializedString(evalModelPath), NativeOnnxValueHelper.GetPlatformSerializedString(optimizerModelPath));
         }
 
         /// <summary>
-        /// Creates TrainingSession from the model and checkpoint in <paramref name="state"/>.
+        /// Create a training session that can be used to begin or resume training.
+        ///
+        /// This constructor instantiates the training session based on the env and session options provided that can
+        /// begin or resume training from a given checkpoint state for the given onnx models.
+        /// The checkpoint state represents the parameters of the training session which will be moved
+        /// to the device specified by the user through the session options (if necessary).
         /// </summary>
-        /// <param name="state">Model checkpoint loaded into <see cref="CheckpointState"/>.</param>
-        /// <param name="trainModelPath">Specify path to training model graph.</param>
-        /// <param name="optimizerModelPath">Specify path to optimizer model graph.</param>
+        /// <param name="state">Training states that the training session uses as a starting point for training.</param>
+        /// <param name="trainModelPath">Model to be used to perform training.</param>
+        /// <param name="optimizerModelPath">Model to be used to perform gradient descent.</param>
         public TrainingSession(CheckpointState state, string trainModelPath, string optimizerModelPath)
         {
             Init(null, state, NativeOnnxValueHelper.GetPlatformSerializedString(trainModelPath), null, NativeOnnxValueHelper.GetPlatformSerializedString(optimizerModelPath));
         }
 
         /// <summary>
-        /// Creates TrainingSession from the model and checkpoint in <paramref name="state"/>.
+        /// Create a training session that can be used to begin or resume training.
+        ///
+        /// This constructor instantiates the training session based on the env and session options provided that can
+        /// begin or resume training from a given checkpoint state for the given onnx models.
+        /// The checkpoint state represents the parameters of the training session which will be moved
+        /// to the device specified by the user through the session options (if necessary).
         /// </summary>
-        /// <param name="state">Model checkpoint loaded into <see cref="CheckpointState"/>.</param>
-        /// <param name="trainModelPath">Specify path to training model graph.</param>
+        /// <param name="state">Training states that the training session uses as a starting point for training.</param>
+        /// <param name="trainModelPath">Model to be used to perform training.</param>
         public TrainingSession(CheckpointState state, string trainModelPath)
```

Comment:
what can we use this API for? Dont we always need optimizer model?

Reply:
I agree with you. I am not sure why we have this. I can probably remove this function altogether.